code/modules/projectiles/guns/energy/lawgiver.dm

```diff
@@ -1,94 +1,179 @@
 /obj/item/weapon/gun/energy/lawgiver
 	name = "Lawgiver Mk II"
-	desc = "A highly advanced firearm for the modern police force. It has multiple voice-activated firing modes."
 	icon_state = "lawgiver"
 	item_state = "gun"
-	fire_sound = 'sound/weapons/Gunshot_smg.ogg'
-	fire_delay = 3
-	recoil = 1
-
-
-	projectile_type = /obj/item/projectile/bullet/pistol
 	origin_tech = "combat=6;magnets=5"
+	projectile_type=/obj/item/projectile/bullet/pistol
+	fire_sound='sound/weapons/Gunshot_smg.ogg'
+	sel_mode = 1
+	desc = "A highly advanced firearm for the modern police force. It has multiple voice-activated firing modes."
+	var/dna	= null//dna-locking the firearm
+	var/emagged = 0 //if the gun is emagged or not
+	var/owner_name = null //Name of the (initial) owner
+	var/message = null //Message that should be played to the user
+	var/message_delay = 100 //Delay of the message_delay
+	var/message_enabled = 0 //If playing the message should be enabled
+	var/message_disable = 0 //If the loop should be stopped
+	var/default_desc = "A highly advanced firearm for the modern police force. It has multiple voice-activated firing modes."
 
-	var/ammomode = 0
-	var/dna	//dna-locking the firearm
+	firemodes = list(
+		list(
+			name="singleshot",
+			charge_cost=50,
+			fire_delay=3,
+			recoil=1
+		),
+		list(
+			name="rapidfire",
+			charge_cost=150,
+			fire_delay=3,
+			recoil=1,
+			burst=3,
+			move_delay=4,
+			accuracy = list(0,-1,-1,-2,-2),
+			dispersion = list(0.0, 0.6, 1.0)
+		),
+		list(
+			name="highex",
+			charge_cost=300,
+			fire_delay=6,
+			recoil=3
+		),
+		list(
+			name="stun",
+			charge_cost=50,
+			fire_delay=4,
+			recoil=0
+		),
+		list(
+			name="hotshot",
+			charge_cost=200,
+			fire_delay=4,
+			recoil=3
+		),
+		list(
+			name="armorpiercing",
+			charge_cost=300,
+			fire_delay=6,
+			recoil=3
+		),
+		list(
+			name="pellets",
+			charge_cost=300,
+			fire_delay=6,
+			recoil=3
+		),
+	)
 
-	proc/hear(message, source as mob)
-		return
+	/obj/item/weapon/gun/energy/lawgiver/proc/play_message()
```

Comment:
Actually, just noticed something. All lines starting from here should be tabbed back by one. Previously it was set up to use partial path names, but you're using full path names, so you start from 0-tabs.